### PR TITLE
Add "needs testing" label if contributor comments "ready for testing" 

### LINKED ISSUE #7479: Add "needs testing" label if contributor comments "ready for testing"
## Motivation
Since we have a growing number of contributions and since open-source contributors do not have triage permissions, we might miss to add "needs testing" label and the PR might become stale without being noticed.

## Solution

- [ ] Add an auto label workflow for adding the "needs testing" label.

The needs testing label can be added if either of the following conditions are met:

- User comments "ready for testing" and the PR is not a draft PR
- PR opened and is not a draft PR
- PR changed from draft status to open.

- [ ] When "changes required" label is added by a user, add a comment in the PR to remind the contributor on how to add the "needs testing" label by commenting "ready for testing".

Feel free to suggest alternatives or add on to this.

### PR DESCRIPTION
**Pull request by saloni0419**
## Proposed Changes
Adds "needs testing" label if contributor comments "ready for testing" 

- Fixes #7479 

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

### CODE REVIEW

**Comment by saloni0419** [comment]
@rithviknishad @aparnacoronasafe @Ashesh3  please check this pr.

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
Marking it tested as nothing to test from QA side, @khavinshankar review and merge it

**Review by khavinshankar — APPROVED** [review]

File: .github/workflows/auto-testing-label.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        uses: actions/github-script@v5
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const pr = context.payload.pull_request;
+            const isDraft = pr.draft;
+            const isReadyForTestingComment = context.payload.comment?.body.includes('ready for testing');
+            const isChangesRequired = context.payload.review?.state === 'changes_requested';
+
+            if ((isReadyForTestingComment && !isDraft) || (!isDraft && pr.draft_changed) || isChangesRequired) {
```
Comment (khavinshankar) [inline .github/workflows/auto-testing-label.yml]:
```suggestion
            if ((isReadyForTestingComment && !isDraft) || (!isDraft && pr.draft_changed)) {
```

should the `need testing` label be added if changes are requested?

File: .github/workflows/auto-testing-label.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+              await github.issues.addLabels({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                issue_number: pr.number,
+                labels: ['needs testing']
+              });
+            }
+
+            if (isChangesRequired) {
```
Comment (khavinshankar) [inline .github/workflows/auto-testing-label.yml:36]: Also we could remove `need testing` label when changes are requested, not mandatory but good to have

cc: @nihal467

**Review by khavinshankar — CHANGES_REQUESTED** [review]

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**